repoman/git/depot_operations.py

```diff
@@ -42,56 +42,39 @@ def check_changeset_availability(self, path, changesets):
         """
         Inherited method :func:`~DepotOperations.check_changeset_availability`
         """
+        # TODO: Check if changesets are really in path
         return changesets
 
     def grab_changesets(self, path, url, changesets):
         """
         Inherited method :func:`~DepotOperations.grab_changesets`
         """
         logger.debug('Grabbing changesets from %s to %s' % (url, path))
-        git_repo = pygit2.Repository(path)
-        try:
-            origin = self._set_origin_source(git_repo, url)
-            result = origin.fetch()
-            logger.debug('GIT Done grabbing changesets (%s)' % (result))
-            self._save_state(path)
-        except pygit2.GitError as e:
-            logger.exception('Error Grabbing changesets: %s' % e)
-            return False
+        sh.git("fetch", url, "+refs/heads/*:refs/remotes/origin/*", _cwd=path)
```

Comment:
Why was it set previously the refs as `+refs/*:refs/*` ? Overwrite all references in each fetch?
I didn't have any reason for that, and it would be better to use refs git as you set it here (refs/remotes/origin).

Reply:
Althoug it's weird for git, maybe it'd be more consistent with previous behaviour to fetch with `+refs/*:refs/*`, yes, I think that there are other parts of the code that also expect that, and there are still some tests failing with issues with branches.
It's not possible to fetch to an attached branch, but we can detach before fetch. I'll try to change it back to `+refs/*:refs/*`